### PR TITLE
Update podspec to ReactiveSwift 5.0

### PR DESCRIPTION
**Pull request by Marcocanc**
Forgot to update the podspec in #3646

### CODE REVIEW

File: ReactiveCocoa.podspec [hunk 1]
```diff
@@ -24,7 +24,7 @@ Pod::Spec.new do |s|
   s.watchos.source_files = "ReactiveCocoa/WatchKit/*.{swift}"
   s.module_name = 'ReactiveCocoa'
 
-  s.dependency 'ReactiveSwift', '~> 4.0'
+  s.dependency 'ReactiveSwift', '~> 5.0'
 
   s.pod_target_xcconfig = { "OTHER_SWIFT_FLAGS[config=Release]" => "$(inherited) -suppress-warnings" }
   s.swift_version = '4.1.2'
```
Comment (ikesyo) [inline ReactiveCocoa.podspec]: This should be `4.2`.

File: ReactiveCocoa.podspec [hunk 2]
```diff
@@ -24,7 +24,7 @@ Pod::Spec.new do |s|
   s.watchos.source_files = "ReactiveCocoa/WatchKit/*.{swift}"
   s.module_name = 'ReactiveCocoa'
 
-  s.dependency 'ReactiveSwift', '~> 4.0'
+  s.dependency 'ReactiveSwift', '~> 5.0'
```
Comment (ikesyo) [inline ReactiveCocoa.podspec]: FYI ReactiveSwift 5.0 is not published to CocoaPods yet. Please wait for a while.

File: .travis.yml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 language: objective-c
-osx_image: xcode10.1
+osx_image: xcode10.2
```
Comment (ikesyo) [inline .travis.yml]: We need https://github.com/ReactiveCocoa/ReactiveCocoa/pull/3647 before using Xcode 10.2 image.

File: ReactiveCocoa.podspec (hunk 2 above)
Comment (ikesyo) [reply ReactiveCocoa.podspec]: That has been published.

File: .travis.yml (hunk 3 above)
Comment (mdiep) [reply .travis.yml]: It has been merged now. ✨

File: ReactiveCocoa.podspec (hunk 1 above)
Comment (Marcocanc) [reply ReactiveCocoa.podspec]: Should we merge one of the Swift 4.2 branches first then?

Comment (ikesyo) [reply ReactiveCocoa.podspec]: Ah okay this is not required, I'm sorry for confusing you!

(Both Swift 4.2 compiler and Swift 5 compiler support Swift 4 mode, so this is still valid)

**Pull request merged**